### PR TITLE
Unify settings [OP-42]

### PR DESCRIPTION
**Pull request by caneruguz**
This PR removes all layer specific settings and adds a single settings component that takes its setting definitions from the form:[] field in the layer data.

### CODE REVIEW

File: app/components/layer-advanced-settings/template.hbs [hunk 1]
```diff
@@ -4,12 +4,12 @@
         <div class="row">
```
Comment (Rytiggy) [inline app/components/layer-advanced-settings/template.hbs]: Are we not removing the advanced layers settings?

File: app/components/add-layer/component.js [hunk 2]
```diff
@@ -55,7 +55,6 @@ export default Ember.Component.extend({
                 case 'advanced': // Update when link is ready
```
Comment (Rytiggy) [inline app/components/add-layer/component.js]: I don't think this comment was from this PR but should we remove it?

File: app/components/add-layer/component.js [hunk 3]
```diff
@@ -72,7 +71,6 @@ export default Ember.Component.extend({
                 case 'image': // Update when link is ready
```
Comment (Rytiggy) [inline app/components/add-layer/component.js]: Should we remove this comment as well

File: app/components/layer-advanced-settings/template.hbs (hunk 1 above)
Comment (caneruguz) [inline app/components/layer-advanced-settings/template.hbs]: ah forgot that, doing it now

File: app/components/add-layer/component.js (hunk 2 above)
Comment (caneruguz) [inline app/components/add-layer/component.js]: Yeah all these comments are from when link was not a layer I think. Removing them now.

**Pull request merged**